lib/engine/tile.rb

```diff
@@ -395,6 +395,14 @@ def city_town_edges
       ct_edges.values
     end
 
+    def city_town_edges_are_subset_of?(other_cte)
+      cte = city_town_edges
+      ALL_EDGES.any? do |rotation|
+        rotated = other_cte.map { |city| city.map { |edge| rotate(edge, rotation) } }
```

Comment:
why are you calling a map of maps, isn't this returning an array of arrays?

Reply:
Yes, `city_town_edges` (which existed before this PR) returns an array for arrays - each item in the outer array is a city on the tile, each item in the inner array is an edge that city is connected to.

For example, for this tile:
![image](https://user-images.githubusercontent.com/8403378/200279307-7a5d6cdb-62e6-49c1-a87c-717319d9cce9.png)
it returns `[[0, 3], [2, 4]]`

The map of maps is to translate this result for each possible rotation.